### PR TITLE
change dockerfile links, add tzdata to fulltext, migrate healthchecks to nc

### PR DESCRIPTION
**Pull request by Zoey2936**
Signed-off-by: Zoey <zoey@z0ey.de>
for #1617

### CODE REVIEW

**Comment by Zoey2936** [comment]
@szaimen as this seems to not help #1617 I would say that I remove the /dev/null part and migrate the other healthchecks to netcat

**Comment by szaimen** [comment]
Thanks a lot Zoey! :)

Based on https://github.com/nextcloud/all-in-one/discussions/1617#discussioncomment-4510642, do you still think that this PR is needed? 


(I have not seen any healthcheck log geeting written to the container log so this is not nwcessarily needed imho...)

**Comment by szaimen** [comment]
> @szaimen as this seems to not help #1617 I would say that I remove the /dev/null part and migrate the other healthchecks to netcat

All right, that sounds good to me 👍

**Comment by Zoey2936** [comment]
@szaimen I think you can now have a look

File: Containers/fulltextsearch/Dockerfile [hunk 1]
```diff
@@ -3,4 +3,12 @@ FROM elasticsearch:7.17.8
...
-HEALTHCHECK CMD curl -skfI localhost:9200 || exit 1
+RUN set -ex; \
+    \
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+        tzdata \
+    ; \
+    rm -rf /var/lib/apt/lists/*
+
+HEALTHCHECK CMD nc -z localhost 9200 || exit 1
```
Comment (szaimen) [inline Containers/fulltextsearch/Dockerfile:14]: are you sure that nc is included by default?

File: Containers/onlyoffice/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # From https://github.com/ONLYOFFICE/Docker-DocumentServer/blob/master/Dockerfile
 FROM onlyoffice/documentserver:7.2.2.56
 
-HEALTHCHECK CMD curl -skfI localhost || exit 1
\ No newline at end of file
+HEALTHCHECK CMD nc -z localhost 80 || exit 1
```
Comment (szaimen) [inline Containers/onlyoffice/Dockerfile:4]: are you sure that nc is included by default?

Comment (Zoey2936) [reply Containers/onlyoffice/Dockerfile:4]:
```
docker run --rm -it --entrypoint nc onlyoffice/documentserver:7.2.2.56
usage: nc [-46CDdFhklNnrStUuvZz] [-I length] [-i interval] [-M ttl]
          [-m minttl] [-O length] [-P proxy_username] [-p source_port]
          [-q seconds] [-s sourceaddr] [-T keyword] [-V rtable] [-W recvlimit]
          [-w timeout] [-X proxy_protocol] [-x proxy_address[:port]]
          [destination] [port]
```

File: Containers/fulltextsearch/Dockerfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/fulltextsearch/Dockerfile:14]:
```
docker run --rm -it --entrypoint nc elasticsearch:7.17.8
Unable to find image 'elasticsearch:7.17.8' locally
7.17.8: Pulling from library/elasticsearch
eaead16dc43b: Already exists 
e247f12a2b5f: Already exists 
a0495283e709: Already exists 
7c7e458ee599: Already exists 
00ffe59d0e26: Already exists 
aab476f874bd: Already exists 
a20df4c47e64: Already exists 
ef90b430ba5b: Already exists 
79f96f41bfec: Already exists 
Digest: sha256:fdc73b3249c1936f7a277911d58bd10bdd5cf7aae07c1f4d285cf3b7bd18e503
Status: Downloaded newer image for elasticsearch:7.17.8
usage: nc [-46CDdFhklNnrStUuvZz] [-I length] [-i interval] [-M ttl]
          [-m minttl] [-O length] [-P proxy_username] [-p source_port]
          [-q seconds] [-s source] [-T keyword] [-V rtable] [-W recvlimit] [-w timeout]
          [-X proxy_protocol] [-x proxy_address[:port]]           [destination] [port]
```

Comment (szaimen) [reply Containers/fulltextsearch/Dockerfile:14]: okay perfect, thanks! :)

File: Containers/onlyoffice/Dockerfile (hunk 2 above)
Comment (szaimen) [reply Containers/onlyoffice/Dockerfile:4]: okay perfect, thanks! :)

**Review by szaimen — APPROVED** [review]
LGTM, thanks! :)

**Comment by szaimen** [comment]
This is now released with v4.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**